woocommerce/woocommerce-gutenberg-products-block
# Update the Interactivity API files to include latest changes

## What?

Updates all files related to the Interactivity API.

Tracking issue: https://github.com/woocommerce/woocommerce/issues/42486

## Why?

To maintain it updated, and thus be able to start using it in blocks (e.g. the "Add to cart" block).

## How?

The JS runtime code comes from [WordPress/block-interactivity-experiments](https://github.com/WordPress/block-interactivity-experiments), but the PHP one comes from [WordPress/gutenberg](https://github.com/WordPress/gutenberg) (more precisely, from the `interactivity` branch).

All prefixes related to `wp` and `gutenberg` were replaced with `wc` and `woocommerce` ones.

There is also some custom code to register the Interactivity API runtime script.

## Testing instructions

You can check the Simple Price Filter block using the updated Interactivity API in the next PR:
- https://github.com/woocommerce/woocommerce-blocks/pull/9926
